File: src/main/java/io/jjong/algorithm/tree/BinaryTreeNode.java
package io.jjong.algorithm.tree;

/**
 * create on 2022/12/07. create by IntelliJ IDEA.
 *
 * <p> 클래스 설명 </p>
 * <p> {@link } and {@link }관련 클래스 </p>
 *
 * @author Jongsang Han(Henry)
 * @version 1.0
 * @see
 * @since 1.0
 */
public class BinaryTreeNode<T> {
  public T data;
  public BinaryTreeNode<T> left, right;

  public BinaryTreeNode(T data) {
    this.data = data;
  }



}

File: src/main/kotlin/io/jjong/algorithm/tree/BinaryTreeNode.kt
package io.jjong.algorithm.tree

/**
 * 이진 트리 노드. [data]는 값, [left]/[right]는 좌/우 자식(없으면 null).
 */
class BinaryTreeNode<T>(
    var data: T?,
    var left: BinaryTreeNode<T>? = null,
    var right: BinaryTreeNode<T>? = null,
)
